src/pybind/mgr/snap_schedule/fs/schedule.py

```diff
@@ -55,10 +55,7 @@ def parse_timestamp(ts: str) -> datetime:
 def parse_retention(retention: str) -> Dict[str, int]:
     ret = {}
     log.debug(f'parse_retention({retention})')
-    matches = re.findall(r'\d+[a-z]', retention)
-    for m in matches:
-        ret[m[-1]] = int(m[0:-1])
-    matches = re.findall(r'\d+[A-Z]', retention)
+    matches = re.findall(r'\d+[nmhdwMy]', retention)
```

Comment:
I don't have any objections from this change, but just wanted to understand if this request originated from our upstream users?

Reply:
No, this change isn't derived out any upstream request.
This change has been proposed to avoid accepting meaningless period specifiers and then scamper to fix problems arising out of QA tests.
The inspiration was due to negative tests conducted downstream.

Reply:
Fair enough. We need to document this in docs and in PendingReleaseNotes.

I'm a bit worried about potential scheduled already used in the field which might stop working post this patch. Should we auto-convert them to the appropriate schedules and then probably after a couple of releases do away with the auto-conversion too?

Reply:
I'll make changes to the qa tests to pass bad retention spec with a `CommandFailedError` after the `fs volume ls` stall issue solution has been merged so that I work on a stable baseline.